class/actions_subtotal.class.php

```diff
@@ -1162,30 +1177,74 @@ function isModSubtotalLine(&$parameters, &$object) {
 	function pdf_getlineqty($parameters=array(), &$object, &$action='') {
 		global $conf,$hideprices;
 
+        $i = intval($parameters['i']);
+        $line = $object->lines[$i];
+
 		if($this->isModSubtotalLine($parameters,$object) ){
-			$this->resprints = ' ';
+            if ($this->subtotal_sum_qty_enabled === true) {
+                $line_qty = intval($line->qty);
+
+                if ($line_qty < 50) {
+                    // it's a title level (init level qty)
+                    $subtotal_level = $line_qty;
+                    $this->subtotal_level_cur = $subtotal_level;
+                    TSubtotal::setSubtotalQtyForObject($object, $subtotal_level, 0);
+
+                    // not show qty for title lines
+                    $this->resprints = '';
+
+                    return 1;
+                } elseif ($line_qty > 50) {
+                    // it's a subtotal level (show level qty and reset)
+                    $subtotal_level = 100 - $line_qty;
+                    $level_qty_total = $object->TSubtotalQty[$subtotal_level];
+                    TSubtotal::setSubtotalQtyForObject($object, $subtotal_level, 0);
+
+                    // show quantity sum only if it's a subtotal line (level)
+                    $line_show_qty = TSubtotal::showQtyForObjectLine($line, $this->subtotal_show_qty_by_default);
+                    if ($line_show_qty === false) {
+                        $this->resprints = '';
+                    } else {
+                        $this->resprints = $level_qty_total;
+                    }
 
-			if((float)DOL_VERSION<=3.6) {
-				return '';
-			}
-			else if((float)DOL_VERSION>=3.8) {
-				return 1;
-			}
+                    return 1;
+                } else {
+                    // not show qty for text line
+                    $this->resprints = '';
+                    return 1;
+                }
+            }
+            else {
+                $this->resprints = ' ';
 
-		}
-		elseif(!empty($hideprices)) {
-			$this->resprints = $object->lines[$parameters['i']]->qty;
-			return 1;
-		}
-		elseif (!empty($conf->global->SUBTOTAL_IF_HIDE_PRICES_SHOW_QTY))
-		{
-			$hideInnerLines = GETPOST('hideInnerLines', 'int');
-			$hidedetails = GETPOST('hidedetails', 'int');
-			if (empty($hideInnerLines) && !empty($hidedetails))
-			{
-				$this->resprints = $object->lines[$parameters['i']]->qty;
-			}
-		}
+                if ((float)DOL_VERSION <= 3.6) {
+                    return '';
+                } else if ((float)DOL_VERSION >= 3.8) {
+                    return 1;
+                }
```

Comment:
does not support anymore Dolibarr version less than  Dolibarr 6.0
```suggestion
                    return 1;
```